tests/test_plot_manager.py

```diff
@@ -0,0 +1,185 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from .common import test_result_collector as trc
+from .common.test_utils import convert_to_bytes, default_encode, ROOT_DIR
+from .mocks.mock_config import MockConfig
+
+from model_analyzer.cli.cli import CLI
+from model_analyzer.config.input.config_command_analyze \
+    import ConfigCommandAnalyze
+
+from model_analyzer.plots.plot_manager import PlotManager
+from model_analyzer.result.result_manager import ResultManager
+from model_analyzer.state.analyzer_state_manager import AnalyzerStateManager
+
+from filecmp import cmp
+from shutil import rmtree
+from unittest.mock import MagicMock, patch
+
+import json
+
+
+class TestPlotManager(trc.TestResultCollector):
+
+    def setUp(self):
+        self._create_single_model_result_manager()
+        self._create_multi_model_result_manager()
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_single_model_summary_plots_against_golden(self):
+        """
+        Match the summary plots against the golden versions in
+        tests/common/single-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._single_model_config,
+            result_manager=self._single_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
+        with open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.json", "wb") as f:
+            f.write(
+                convert_to_bytes(
+                    json.dumps(plot_manager.to_dict(), default=default_encode)))
+
+        with open(f"{ROOT_DIR}/single-model-ckpt/plot_manager.json", "rb") as f:
+            golden_plot_manager_dict = json.load(f)
+
+        golden_plot_manager = PlotManager.from_dict(
+            golden_plot_manager_dict, self._single_model_config,
+            self._single_model_result_manager)
+
+        golden = golden_plot_manager._simple_plots['add_sub'][
+            'throughput_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['throughput_v_latency']
+        self._compare_plot_data(golden, plot)
+
+        golden = golden_plot_manager._simple_plots['add_sub'][
+            'gpu_mem_v_latency']
+        plot = plot_manager._simple_plots['add_sub']['gpu_mem_v_latency']
+        self._compare_plot_data(golden, plot)
+
+    def test_multi_model_summary_plots_against_golden(self):
+        """
+        Match the summary plots against the golden versions in
+        tests/common/multi-model-ckpt
+        """
+        plot_manager = PlotManager(
+            config=self._multi_model_config,
+            result_manager=self._multi_model_result_manager)
+
+        plot_manager.create_summary_plots()
+
+        # Uncomment these lines to create a new output dump to compare against
```

Comment:
Should the code after this be commented out?

Reply:
Yes, good catch.